diffrax/_solver/langevin_srk.py

```diff
@@ -0,0 +1,325 @@
+import abc
+from typing import Any, Callable, Generic, TypeVar
+
+import equinox as eqx
+import equinox.internal as eqxi
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from equinox import AbstractVar
+from jax import vmap
+from jax._src.tree_util import PyTreeDef
+from jaxtyping import Array, ArrayLike, PyTree
+
+from .._custom_types import (
+    AbstractBrownianIncrement,
+    BoolScalarLike,
+    DenseInfo,
+    RealScalarLike,
+)
+from .._local_interpolation import LocalLinearInterpolation
+from .._solution import RESULTS
+from .._term import (
+    _LangevinDiffusionTerm,
+    _LangevinDriftTerm,
+    _LangevinTuple,
+    _LangevinX,
+    AbstractTerm,
+    MultiTerm,
+    WrapTerm,
+)
+from .base import AbstractItoSolver, AbstractStratonovichSolver
+
+
+_ErrorEstimate = TypeVar("_ErrorEstimate", None, _LangevinTuple)
+_LangevinArgs = tuple[_LangevinX, _LangevinX, Callable[[_LangevinX], _LangevinX]]
+
+
+def get_args_from_terms(
+    terms: MultiTerm[tuple[AbstractTerm[Any, RealScalarLike], AbstractTerm]],
+) -> _LangevinArgs:
+    drift, diffusion = terms.terms
+    if isinstance(drift, WrapTerm):
+        unwrapped_drift = drift.term
+        assert isinstance(diffusion, WrapTerm)
+        unwrapped_diffusion = diffusion.term
+        assert isinstance(unwrapped_drift, _LangevinDriftTerm)
+        assert isinstance(unwrapped_diffusion, _LangevinDiffusionTerm)
+        gamma = unwrapped_drift.gamma
+        u = unwrapped_drift.u
+        f = unwrapped_drift.grad_f
+    else:
+        assert isinstance(drift, _LangevinDriftTerm)
+        assert isinstance(diffusion, _LangevinDiffusionTerm)
+        gamma = drift.gamma
+        u = drift.u
+        f = drift.grad_f
```

Comment:
I think you can merge these two pieces. (Just set `drift = drift.term` and `diffusion = diffusion.term` on the `WrapTerm` branch.)

Will this do the right thing when solved backward in time?

Reply:
Yes right the two branches were separate because I previously didn't have this as a separate function and I wanted to keep hold on the "wrapped" versions, precisely for the sake of direction. Because I use `term.contr` both for `h` and `levy` I am pretty convinced it should do the right thing. I'll write a test tomorrow.